optuna/study.py

```diff
@@ -326,11 +346,15 @@ def optimize(
 
                     parallel(
                         delayed(self._optimize_sequential)
-                        (func, 1, timeout, catch, callbacks, gc_after_trial, time_start)
+                        (
+                            func, 1, timeout, catch, callbacks, gc_after_trial,
+                            time_start, progress_bar
```

Comment:
I"m a bit skeptical about passing a progress bar (that's either picklable or not, depending on a boolean argument to it's `__init__`) to each worker. It seems tricky with joblib, unfortunately, but with e.g. multiprocessing I believe you could use `imap` to get intermediate worker results to update the progress bar only in the main process, as workers (i.e. trials) finish.

Reply:
I tested as an alternative logging + tqdm + multiprocessing using joblib's experimental API for registering custom backends (with callbacks). It probably won't render as expected as long as we're using a [`logging.StreamHandler` as the base handler](https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes) however.

https://gist.github.com/hvy/8b80c2cedf02b15c24f85d1fa17ebe02

Reply:
We can probably discuss this later, e.g. in https://github.com/optuna/optuna/issues/876.

Reply:
I feel like it might be better to access `_optimize_parallel` case in separate PRs.